src/components/miscellaneous/ShareButton.tsx

```diff
@@ -0,0 +1,40 @@
+// Material UI Components
+import Button from '@material-ui/core/Button';
+
+import useShare from 'use-share';
+
+export type ShareProps = {
+  title?: string;
+  type: 'event' | 'news' | 'jobpost' | 'pages';
+  id: number | string;
+};
+
+const ShareButton = ({ type, id, title }: ShareProps) => {
+  let typeToURL = '';
+
+  switch (type) {
+    case 'event':
+      typeToURL = 'a';
+      break;
+    case 'news':
+      typeToURL = 'n';
+      break;
+    case 'jobpost':
+      typeToURL = 'k';
+      break;
+    case 'pages':
+      typeToURL = 'om';
+      break;
+  }
+  const { share, hasShared } = useShare({
+    title: `${title}`,
+    url: `https://s.tihlde.org/${typeToURL}/${id}/`,
+  });
+  return (
+    <Button disabled={hasShared} onClick={share}>
```

Comment:
Denne bør kunne styles fra der den brukes som props, i tillegg til å ha litt mer styling by default.
```suggestion
    <Button disabled={hasShared} onClick={share} variant='outlined' {...props}>
```